scripts/run-related-test.mjs

```diff
@@ -0,0 +1,42 @@
+/**
+ * This script finds all `.tests` files under the paths and reads lines from these files.
+ * The script is useful to find all related test cases for a given code change.
+ * Usage: `node scripts/run-related-test.mjs <path1> <path2> ...`
+ */
+
+import { promisify } from 'node:util'
+import { exec as execOrg } from 'node:child_process'
+import { readFile } from 'node:fs/promises'
+import { dirname, join } from 'node:path'
+
+const exec = promisify(execOrg)
+
+/**
+ * Get all changed files from git under the packages directory
+ * @returns {Promise<string[]>} - List of changed files
+ */
+async function getChangedFilesFromPackages(baseBranch = 'origin/canary') {
+  const { stdout } = await exec(`git diff --name-only ${baseBranch}`)
```

Comment:
We probably need to copy the diff detection from https://github.com/vercel/next.js/blob/6a1e70ae55d56a06f51c36ea27b5918dfebb3e99/scripts/run-for-change.js#L73-L74

as otherwise canary isn't cloned/available for diffing by default

Reply:
It was added as part of the workflow in https://github.com/vercel/next.js/pull/63763/files#diff-f0875f562711cdd6393f47891a7b6c7ca3bc34dfd63675bfb2b2275bb0a870d1R50

Typically, if you would run this script locally, you already have the `canary` branch set up.